Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,158 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+            foreach (List<Point> pts in pointLists)
+            {
+                bool isContained = false;
+                if (acceptPartialContainment)
+                    isContained = pts.Any(point => IsContaining(closedVolume, planes, boundingBox, point, acceptOnEdges));
+                else
+                    isContained = pts.All(point => IsContaining(closedVolume, planes, boundingBox, point, acceptOnEdges));
+                areContained.Add(isContained);
+            }
+
+            return areContained;
+        }
+
+        /***************************************************/
+        /**** Private Methods - Closed Volume Polylines ****/
+        /***************************************************/
+
+        [Description("Checks if a point lies within a collection of polylines describing a closed volume using their primitive planes and bounds.")]
+        [Input("closedVolume", "A collection of Polylines outlining each of the faces of the closed volume to check with.")]
+        [Input("planes", "Planes corresponding to each panel for intersection calculations.")]
+        [Input("boundingBox", "The bounding box of the collection of panels.")]
+        [Input("point", "The point to check to see if it is contained within the bounds of the panels.")]
+        [Input("acceptOnEdges", "Decide whether to allow the point to sit on the edge of the panel, default false.")]
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> closedVolume, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (planes.Any(x => x == null))
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a provided plane is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            Vector rVec;
+            bool isInPlane = false;
+            do
+            {
+                int seed = 1;
+                 rVec = BH.Engine.Geometry.Create.RandomVector(seed); 
+                
+                isInPlane = false;
+                foreach (Plane p in planes)
+                    isInPlane = isInPlane || rVec.IsInPlane(p, tolerance);
+
+                if (isInPlane)
+                {
+                    seed += 1;
+                }
```

Comment:
You need to normalise the random vector.

Besides that I could not stop myself from rewriting the code according to my own likings 😉 But it is a matter of personal preference, so please feel free to ignore 👍 
```suggestion
                int seed = 1;
                rVec = BH.Engine.Geometry.Create.RandomVector(seed).Normalise();                 
                isInPlane = planes.Any(p => rVec.IsInPlane(p, tolerance));

                if (isInPlane)
                    seed ++;
```

Reply:
Modified per suggestion.